src/v/cloud_storage/remote.cc

```diff
@@ -975,27 +995,95 @@ ss::future<upload_result> remote::delete_object(
     co_return *result;
 }
 
+template<typename R>
+requires std::ranges::range<R>
+         && std::same_as<
+           std::ranges::range_value_t<R>,
+           cloud_storage_clients::object_key>
 ss::future<upload_result> remote::delete_objects(
   const cloud_storage_clients::bucket_name& bucket,
-  std::vector<cloud_storage_clients::object_key> keys,
+  R keys,
   retry_chain_node& parent) {
     ss::gate::holder gh{_gate};
+    retry_chain_logger ctxlog(cst_log, parent);
 
     if (keys.empty()) {
-        vlog(cst_log.info, "No keys to delete, returning");
+        vlog(ctxlog.info, "No keys to delete, returning");
         co_return upload_result::success;
     }
 
+    vlog(ctxlog.debug, "Delete objects count {}", keys.size());
+
     if (!is_batch_delete_supported()) {
         co_return co_await delete_objects_sequentially(
-          bucket, std::move(keys), parent);
+          bucket, std::forward<R>(keys), parent);
```

Comment:
I don't think that forwarding keys here has any difference since it's always an l-value in this case.